### PR TITLE
Badge component: Fix Storybook url link

### PR DESCRIPTION
**Pull request by t-hamano**
## What?

Fix an issue where the Storybook link in the `Badge` component README was not linking to an actual story:

https://github.com/user-attachments/assets/106a18ae-e6c1-4904-ad4f-3ac7d38ee694

## Why?

This is because the URL rules are different when automatically generating a README and when using Storybook.

## How?

By adding an `id` field to the storybook meta,  change the story URL of the `Badge` component as follows:

- From: https://wordpress.github.io/gutenberg/?path=/docs/components-containers-badge--docs
- To: https://wordpress.github.io/gutenberg/?path=/docs/components-badge--docs

### Testing Instructions for Keyboard

- Run `npm run storybook:dev`
- The story URL for the `Badge` component should be `http://localhost:50240/?path=/docs/components-badge--docs`.

### CODE REVIEW

File: bin/api-docs/gen-components-docs/markdown/props.mjs [hunk 1]
```diff
@@ -48,4 +48,3 @@ export function generateMarkdownPropsJson( props, { headingLevel = 2 } = {} ) {
 
 	return [ { [ `h${ headingLevel }` ]: 'Props' }, ...propsJson ];
 }
-
```
Comment (t-hamano) [inline bin/api-docs/gen-components-docs/markdown/props.mjs:51]: This change is to fix a warning in prettier (Two newlines).

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,10 @@
 
 -   `SlotFill`: rewrite the non-portal version to use `observableMap` ([#67400](https://github.com/WordPress/gutenberg/pull/67400)).
 
+### Documentation
+
+-   `Badge`: Fix Storybook url link ([#68077](https://github.com/WordPress/gutenberg/pull/68077)).
```
Comment (tyxla) [inline packages/components/CHANGELOG.md]: Doesn't need a CHANGELOG entry IMO, but up to you.

**Review by tyxla — APPROVED** [review]
Thanks @t-hamano 🙌

File: packages/components/CHANGELOG.md (hunk 2 above)
Comment (t-hamano) [reply packages/components/CHANGELOG.md]: I don't have a strong opinion either, I'll delete this and then merge 👍

**Pull request merged**